src/app/teams/teams-view-finances.component.ts

```diff
@@ -125,11 +125,19 @@ export class TeamsViewFinancesComponent implements OnInit, OnChanges {
           date: [ transaction.date ? new Date(new Date(transaction.date).setHours(0, 0, 0)) : new Date(time), CustomValidators.required ]
         },
         {
-          onSubmit: (newTransaction) => this.submitTransaction(newTransaction, transaction).subscribe(() => {
-            this.planetMessageService.showMessage('Transaction added');
+          onSubmit: (newTransaction) => {
+            if (!transaction._id) {
+              this.submitTransaction(newTransaction, transaction).subscribe(() => {
+              this.planetMessageService.showMessage('Transaction Added');
+              });
+            } else {
+              this.submitTransaction(newTransaction, transaction).subscribe(() => {
+              this.planetMessageService.showMessage('Transaction Edited');
+              });
+            }
             this.dialogsFormService.closeDialogsForm();
```

Comment:
This line was called in the subscription, now it is called before the subscription completes.

Please make sure after making changes that this happens after `submitTransaction` completes.